src/jit/namedintrinsiclist.h

```diff
@@ -89,9 +89,22 @@ enum HWIntrinsicFlag : unsigned int
     // the imm intrinsic does not need jumptable fallback when it gets non-const argument
     HW_Flag_NoJmpTableIMM = 0x2000,
 
+    // 64-bit intrinsics
+    // Intrinsics that operate over 64-bit general purpose registers are not supported on 32-bit platform
+    HW_Flag_64BitOnly           = 0x4000,
+    HW_Flag_SecondArgMaybe64Bit = 0x8000,
+
+    // Select base type using the second argument type
+    HW_Flag_BaseTypeFromSecondArg = 0x10000,
+
+    // Specail codegen
+    // the intrinsics need special rules in CodeGen,
+    // but can be table-driven in the front-end
+    HW_Flag_SpecialCodeGen = 0x20000,
+
     // No Read/Modify/Write Semantics
     // the intrinsic does not have read/modify/write semantics and doesn't need
```

Comment:
This is pre-existing, but this comment isn't complete.
I would qualify it with "the intrinsic doesn't have read/modify/write semantics in two-operand form." (or non-AVX, or whatever the appropriate characterization is).